packages/php/email-editor/src/Engine/Renderer/ContentRenderer/Preprocessors/class-spacing-preprocessor.php

```diff
@@ -39,13 +39,13 @@ private function add_block_gaps( array $parsed_blocks, string $gap = '', $parent
 			$parent_block_name = $parent_block['blockName'] ?? '';
 			// Ensure that email_attrs are set.
 			$block['email_attrs'] = $block['email_attrs'] ?? array();
+
 			/**
 			 * Do not add a gap to:
-			 * - the top level blocks - they are post-content, and header and footer wrappers and we don't want a gap between those
 			 * - first child
 			 * - parent block is a buttons block (where buttons are side by side).
 			 */
-			if ( $parent_block && 0 !== $key && $gap && 'core/buttons' !== $parent_block_name ) {
+			if ( 0 !== $key && $gap && 'core/buttons' !== $parent_block_name ) {
```

Comment:
No action required, but I think we should really extract complicated conditions like this into a separate function.

Reply:
The condition here should be simple. IMO the `'core/buttons' !== $parent_block_name ` should not be there and this code shouldn't depend on block names, because it is not extendible for new blocks. This is one of the reasons to look into better architecture for the renderer.